src/main.rs

```diff
@@ -84,8 +84,11 @@ fn ldap_client(
         .set_no_tls_verify(no_tls_verify);
     let mut ldap = LdapConn::with_settings(settings, ldap_url.as_str())
         .context("Failed to connect to LDAP server")?;
-    ldap.simple_bind(bind_dn.as_str(), bind_pw.as_str())
-        .context("Could not bind to LDAP server")?;
+
+    if bind_dn.is_some() && bind_pw.is_some() {
+        ldap.simple_bind(&bind_dn.unwrap(), &bind_pw.unwrap())
+            .context("Could not bind to LDAP server")?;
+    }
```

Comment:
Rather than using `is_some()` and `unwrap()`, we should use `match` here. This is not only more idiomatic, but it allows us to exhaust the possible branches and provide more descriptive errors.

Here's a simple example of what I mean:

```rust
match (bind_dn, bind_pw) {
        (Some(dn), Some(pw)) => println!("bind as {}:{}. Use ldap.simple_bind(...)", dn, pw),
        
        (None, None) => println!("anon bind. Use ldap.bind()"),
        // These last two should raise errors 
        (Some(dn), None) => println!("bind as {} with no pw. Should error.", dn),
        (None, Some(_)) => println!("Invalid! Can't bind with pw only. Should error.")
    }
 ```
Example on playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c12ddb12f80b77d438cc5b043f9e1779

Also, you'll see in the example I note that you should still bind if it's anon. You need to do `ldap.bind()` instead. https://ldap3.readthedocs.io/en/latest/bind.html#anonymous-bind